src/main/java/com/paneedah/mwc/groovyscript/script/CraftingStation.java

```diff
@@ -0,0 +1,195 @@
+        }
+
+        private final ArrayList<CraftingEntry> entries = new ArrayList<>();
+        private CraftingGroup group = GUN;
+
+        /**
+         * Add Ingredient of recipe With ItemStack
+         */
+        public RecipeBuilder addEntry(ItemStack stack){
+            return addEntry(stack, 1);
+        }
+
+        /**
+         * Add Ingredient of recipe With Ore Dictionary Value
+         */
+
+        public RecipeBuilder addOredict(String entry, int count){
+            return addOredict(entry, count, 1);
+        }
+
+        /**
+         * Add Ingredient of recipe With given ItemStack and yield(0.0~1.0)
+         */
+
+        public RecipeBuilder addEntry(ItemStack stack, double yield){
+            this.entries.add(new CraftingEntry(stack.getItem(), stack.getCount(), yield));
+            return this;
+        }
+
+        /**
+         * Add Ingredient of recipe With given Ore Dictionary Entry and yield(0.0~1.0)
+         */
+
+
+        public RecipeBuilder addOredict(String entry, int count, double yield){
+            this.entries.add(new CraftingEntry(OreDictionary.getOres(entry).get(0).getItem(),entry, count, yield));
+            return this;
+        }
+
+        /**
+         * Set Category of Recipe
+         */
+        public RecipeBuilder group(String group){
+            this.group = CraftingGroup.valueOf(group);
+            return this;
+        }
+
+        public RecipeBuilder groupGun(){
+            this.group = GUN;
+            return this;
+        }
+
+        public RecipeBuilder groupNormalAttachment(){
+            this.group = ATTACHMENT_NORMAL;
+            return this;
+        }
+
+        public RecipeBuilder groupModificationAttachment(){
+            this.group = ATTACHMENT_MODIFICATION;
+            return this;
+        }
+
+        public RecipeBuilder groupBullet(){
+            this.group = BULLET;
+            return this;
+        }
+
+        public RecipeBuilder groupMagazine(){
+            this.group = MAGAZINE;
+            return this;
+        }
+
+        public RecipeBuilder groupGranade(){
+            this.group = GRENADE;
+            return this;
+        }
+
+        public RecipeBuilder groupGear(){
+            this.group = GEAR;
+            return this;
+        }
+
+        @Override
+        public void validate(GroovyLog.Msg msg) {
+            validateItems(msg, 0,0,1,1);
+            msg.add(this.entries.isEmpty() ,"Ingredient for Crafting Recipe can not be empty.");
+        }
+
+        /**
+         * and we finalize the recipe and register it!
+         */
+
+        @Override
+        public GSCrafting register() {
+            if(!this.validate()){
+                return null;
+            }
+            GSCrafting recipe = new GSCrafting(this.output.get(0).getItem(), this.group, this.entries.toArray(new CraftingEntry[0]));
+            MWC.MWC.get().STATION.addRecipe(recipe);
+            return recipe;
+        }
+    }
+
+}
```

Comment:
Dangling Javadocs,
Javadocs such as:
```java
        /**
         * and we finalize the recipe and register it!
         */
```
should be comments,
Missing space at the top of the class

Reply:
should be fixed on [1f43aa2](https://github.com/Cubed-Development/Modern-Warfare-Cubed/pull/303/commits/1f43aa26210a81a0627894a9a0dbaa87e0a1da56)